openwisp_monitoring/monitoring/base/models.py

```diff
@@ -753,13 +768,19 @@ def _is_crossed_by(self, current_value, time=None, retention_policy=None):
             return value_crossed
         # tolerance is set, we must go back in time
         # to ensure the threshold is trepassed for enough time
+        # check if alert_on_related_field is present in metric configuration
+        if 'alert_on_related_field' in self.metric.config_dict:
+            alert_on_related_field = [self.metric.config_dict['alert_on_related_field']]
+        else:
+            alert_on_related_field = []
```

Comment:
```suggestion
        # if alert field is supplied, retrieve such field when reading
        # so that we can let the system calculate the threshold on it
        extra_values = []
        if self.metric.alert_on_related_field:
           extra_values = [self.metric.alert_field]
```